src/main/kotlin/io/perfometer/runner/CoroutinesScenarioRunner.kt

```diff
@@ -35,7 +46,10 @@ internal class CoroutinesScenarioRunner(
 
     override suspend fun runStep(step: HttpStep) {
         when (step) {
-            is RequestStep -> executeHttp(step)
+            is RequestStep -> executeHttp(
+                coroutineContext[CoroutineHttpClient.Key]?.httpClient ?: throw IllegalStateException(),
```

Comment:
I'd prefer to move this to a separate method - e.g. `httpClient()` or create a private property inside the class that would execute this logic.

Reply:
done